### PR TITLE
delay application of select2

### PR DESCRIPTION
**Pull request by snopoke**
http://manage.dimagi.com/default.asp?244947

Was able to reproduce locally and found that if I ran `$('#report_filter_group').select2();` in the console after the filters had re-loaded then the group options were correctly rendered.

Adding the delay fixed it for me locally, not sure if this is the best solution @dimagi/js-team?

### CODE REVIEW

File: corehq/apps/reports/templates/reports/filters/multi_option.html [hunk 1]
```diff
@@ -28,7 +28,9 @@
         });
     });
     {% if not endpoint %}
-        $('#{{ css_id }}').select2();
+        _.delay(function() {
+            $('#{{ css_id }}').select2();
```
Comment (millerdev) [inline corehq/apps/reports/templates/reports/filters/multi_option.html]: When does this bit of JS get executed? Is it before or after DOM ready or unknown?

**Comment by orangejenny** [comment]
I think moving the `select2` call into the document ready handler directly above it will also fix this problem (tried it out locally) and makes more sense when you're reading the code.

**Comment by snopoke** [comment]
@orangejenny thanks, I hadn't noticed that. Updated to move the `select2` calls inside the document ready handler.

**Review by millerdev — APPROVED** [review]

**Review by esoergel — APPROVED** [review]
Much easier to read like this: https://github.com/dimagi/commcare-hq/pull/14555/files/?w=1
Killing build.

**Pull request merged**